data_diff/__init__.py

```diff
@@ -85,6 +89,8 @@ def diff_tables(
         max_threadpool_size (int): Maximum size of each threadpool. ``None`` means auto.
                                    Only relevant when `threaded` is ``True``.
                                    There may be many pools, so number of actual threads can be a lot higher.
+        print_stats (bool): Enable/disable printing stats to stdout
```

Comment:
Mention that it consumes the entire iterable before returning. i.e. won't return immediately anymore.